Sprint-3/alarmclock/alarmclock.js

```diff
@@ -1,5 +1,61 @@
-function setAlarm() {}
+function formatTimeUnit(unit) {
+  return unit < 10 ? `0${unit}` : unit;
+}
+
+const heading = document.querySelector("#timeRemaining");
+const stopBtn = document.querySelector("#stop");
+heading.innerText = "Time Remaining: 00:00";
+//===========setAlarm======================
+function setAlarm() {
+  const totalTimeInSeconds = parseInt(
+    document.querySelector("#alarmSet").value
+  );
+  setInitialMinutesAndSeconds(heading, totalTimeInSeconds);
+  countDownEverySeconds(totalTimeInSeconds, heading);
+}
+
+//=============setInitialMinutesAndSeconds=======================
+function setInitialMinutesAndSeconds(heading, totalTimeInSeconds) {
+  const initialMinutes = Math.floor(totalTimeInSeconds / 60);
+  const InitialSeconds = totalTimeInSeconds % 60;
+
+  heading.innerText = `Time Remaining: ${formatTimeUnit(
+    initialMinutes
+  )}:${formatTimeUnit(InitialSeconds)}`;
+}
+
+//================countDownEverySeconds======================
+function countDownEverySeconds(totalTimeInSeconds, heading) {
+  const setIntervalId = setInterval(() => {
+    totalTimeInSeconds--;
+    const totalMinutes = Math.floor(totalTimeInSeconds / 60);
+    const remainingSeconds = totalTimeInSeconds % 60;
+    heading.innerText = `Time Remaining: ${formatTimeUnit(
+      totalMinutes
```

Comment:
I find it a little hard to read this code which is split onto lines like this - I think it would be easier to read either:

All on one line:
```js
heading.innerText = `Time Remaining: ${formatTimeUnit(totalMinutes)}:${formatTimeUnit(remainingSeconds)}`;
```

or pulling out intermediate variables:
```js
const formattedMinutes = formatTimeUnit(totalMinutes);
const formattedSeconds = formatTimeUnit(remainingSeconds);
heading.innerText = `${formattedMinutes}:${formattedSeconds}`;
```

What do you think? Which of the three do you prefer? Which is the most clear?

Reply:
I think I prefer the second approach because with prettier the 1st one would be formated as it it at the moment